pkg/check/check.go

```diff
@@ -1,13 +1,65 @@
 package check
 
 import (
+	"context"
+	"fmt"
+	"net/http"
+	"sort"
+
 	"github.com/alphagov/paas-grafana-annotation-resource/pkg/types"
+	"github.com/grafana-tools/sdk"
 )
 
 func Check(req types.CheckRequest) (types.CheckResponse, error) {
+	err := req.Source.Validate()
+
+	if err != nil {
+		return types.CheckResponse{}, err
+	}
+
+	params := []sdk.GetAnnotationsParams{
+		sdk.WithAnnotationType(),
+	}
+	for _, tag := range req.Source.Tags {
+		params = append(params, sdk.WithTag(tag))
+	}
+
+	var client *sdk.Client
+	if req.Source.APIToken != "" {
+		client = sdk.NewClient(req.Source.URL, req.Source.APIToken, http.DefaultClient)
+	} else {
+		client = sdk.NewClient(req.Source.URL, fmt.Sprintf("%s:%s", req.Source.Username, req.Source.Password), http.DefaultClient)
+	}
+
+	annotations, err := client.GetAnnotations(context.Background(), params...)
+	if err != nil || len(annotations) == 0 {
+		return types.CheckResponse{}, err
+	}
```

Comment:
nit: a comment that "returning empty check response and `err` when `err == nil` is okay" would be good